compiler/rustc_parse/src/parser/mod.rs

```diff
@@ -918,15 +918,27 @@ impl<'a> Parser<'a> {
                                 Ok(t) => {
                                     // Parsed successfully, therefore most probably the code only
                                     // misses a separator.
-                                    expect_err
-                                        .with_span_suggestion_short(
+
+                                    // This is to avoid adding suggestions twice for doc comments in item list.
```

Comment:
Right, so an oversight of mine. Since here it already adds the `missing `,`` suggestion, there's no need to have logic that does additional things.

So remove this check, and don't add an additional suggestion in diagnostics.rs.